### ISSUE TITLE
aria-owns should not be used on the trigger, but on its parent

### ISSUE THREAD

**Issue by MarcoZehe** [cibernox/ember-basic-dropdown#557]
This was originally reported as https://github.com/TryGhost/Ghost/issues/11868.

The problem is that, right now, aria-owns gets set on the same element that the button role is on. This makes the basic-dropdown-content container a child of the button. But buttons by definition can only have text content, or graphics, but no other buttons, links, or other inputs inside it.

Some screen readers like NVDA are quite forgiving, although the announcement of everything being a button in addition to a text field or such, is still annoying. Other screen readers, like Orca on Linux, are not so forgiving, as can be seen in the report against the Ghost project.

I am using a local branch where I try to replace aria-owns with aria-controls to still have a relation that screen readers may use, but in the dummy application at least, content is not being added below that, even though the markup says so, but appended to the end of the document to a basic-dropdown-wormhole div that seems to always be there.

So to always make the dropdown content a sibling of the trigger, aria-owns should be applied to the *parent* of the trigger. If only the ID of the content is specified, what will happen is that first all regular children get added to the accessibility tree as usual, and then the aria-owned ones appended to the end. And this is what we want.

Question is: How should I best do that? @cibernox any suggestions?

**Comment by jayjayjpg** [comment]
+1 on this one! Without the fix @MarcoZehe provided in https://github.com/cibernox/ember-basic-dropdown/pull/558 it is not possible for Ember applications to pass accessibility audits via ember-a11y-testing. Would it be possible to merge this bug fix for an upcoming release?

We could benefit from this fix in our applications which use ember-basic-dropdown as well

**Comment by nwhittaker** [comment]
This behavior appears to have regressed following https://github.com/cibernox/ember-basic-dropdown/pull/626. I see `aria-owns` toggled on the parent of the `[data-test-popout-overflow-menu-trigger]` element now.
